app/javascript/components/mentoring/mentor_inbox.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useReducer } from 'react'
+import { ConversationList } from './mentor_inbox/conversation_list'
+import { TrackFilter } from './mentor_inbox/track_filter'
+
+export function reducer(state, action) {
+  switch (action.type) {
+    case 'page.changed':
+      return { ...state, query: { ...state.query, page: action.payload.page } }
+    case 'track.changed':
+      return {
+        ...state,
+        query: { ...state.query, track: action.payload.track, page: 1 },
+      }
```

Comment:
You might want to add `return state` at the bottom _or_ throw an error (or throw in development, return `state` at the bottom), so that you will gracefully "upgrade" when  there are more actions.